pallets/asset/src/lib.rs

```diff
@@ -1638,10 +1786,14 @@ impl<T: Trait> Module<T> {
             ticker,
         ) + value;
 
-        // Charge the given fee.
-        if let Some(op) = protocol_fee_data {
-            T::ProtocolFee::charge_fee(op)?;
-        }
+        // Charge the fees.
+        let ops: Vec<_> = if let Some(op) = protocol_fee_data {
+            vec![op, ProtocolOp::AssetCheckpoint]
+        } else {
+            vec![ProtocolOp::AssetCheckpoint]
+        };
+        T::ProtocolFee::charge_fees(ops.as_ref())?;
```

Comment:
```suggestion
        let mut ops = arrayvec::ArrayVec::<[_; 2]>::new();
        ops.push(ProtocolOp::AssetCheckpoint);
        ops.extend(protocol_fee_data);
        T::ProtocolFee::charge_fees(&*ops)?;
```
Simpler and removes the allocation.